### PR TITLE
69799 configuring radox component to use gpu node

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

**Comment by satr** [comment]
It adds affinity to the pod template - to select nodes with labels `radix-node-gpu` or/and `radix-node-gpu-count`
image
Examples
```
  "nodeAffinity": {
    "requiredDuringSchedulingIgnoredDuringExecution": {
      "nodeSelectorTerms": [
        {
          "matchExpressions": [
            {
              "key": "radix-node-gpu",
              "operator": "In",
              "values": [
                "nvidia-p100"
              ]
            },
            {
              "key": "radix-node-gpu-count",
              "operator": "Gt",
              "values": [
                "5"
              ]
            }
```
```
  "nodeAffinity": {
    "requiredDuringSchedulingIgnoredDuringExecution": {
      "nodeSelectorTerms": [
        {
          "matchExpressions": [
            {
              "key": "radix-node-gpu",
              "operator": "In",
              "values": [
                "nvidia-v100",
                "nvidia-p100"
              ]
            },
            {
              "key": "radix-node-gpu",
              "operator": "NotIn",
              "values": [
                "nvidia-k80"
              ]
            }
```

File: pkg/apis/deployment/radixjobcomponent.go [hunk 1]
```diff
@@ -100,8 +104,8 @@ func (c *jobComponentsBuilder) buildJobComponent(appJob v1.RadixJobComponent) v1
 	deployJob := v1.RadixDeployJobComponent{
 		Name:                 componentName,
 		Image:                image,
-		Ports:                appJob.Ports,
-		Secrets:              appJob.Secrets,
+		Ports:                appJobComponent.Ports,
+		Secrets:              appJobComponent.Secrets,
 		EnvironmentVariables: variables, // todo: use single EnvVars instead
 		Monitoring:           monitoring,
 		Resources:            resources,
```
Comment (nilsgstrabo) [inline pkg/apis/deployment/radixjobcomponent.go:111]: `node `is not set on RadixDeployJobComponent

Comment (nilsgstrabo) [reply pkg/apis/deployment/radixjobcomponent.go:111]: Also, add tests in radixjobcomponent_test.go to test that node config from RA jobs is correctly set on RD jobs

File: pkg/apis/utils/applicationcomponent_builder.go [hunk 2]
```diff
@@ -138,6 +139,7 @@ func (rcb *radixApplicationComponentBuilder) BuildComponent() v1.RadixComponent
 		Variables:               rcb.variables,
 		Resources:               rcb.resources,
 		AlwaysPullImageOnDeploy: rcb.alwaysPullImageOnDeploy,
+		Node:                    rcb.node,
 	}
 }
```
Comment (nilsgstrabo) [inline pkg/apis/utils/applicationcomponent_builder.go:145]: Missing WithNode

File: pkg/apis/utils/componentenvironment_builder.go [hunk 3]
```diff
@@ -77,6 +78,7 @@ func (ceb *radixEnvironmentConfigBuilder) BuildEnvironmentConfig() v1.RadixEnvir
 		Replicas:                ceb.replicas,
 		Resources:               ceb.resources,
 		VolumeMounts:            ceb.volumeMounts,
+		Node:                    ceb.node,
 		AlwaysPullImageOnDeploy: ceb.alwaysPullImageOnDeploy,
 		RunAsNonRoot:            ceb.runAsNonRoot,
 	}
```
Comment (nilsgstrabo) [inline pkg/apis/utils/componentenvironment_builder.go:84]: Missing WithNode

**Review by nilsgstrabo — COMMENTED** [review]
See comments + add test for Node in pipeline/steps/deploy_test

File: pkg/apis/utils/componentenvironment_builder.go (hunk 3 above)
Comment (satr) [reply pkg/apis/utils/componentenvironment_builder.go:90]: Fixed

File: pkg/apis/utils/applicationcomponent_builder.go (hunk 2 above)
Comment (satr) [reply pkg/apis/utils/applicationcomponent_builder.go:151]: Fixed

File: pkg/apis/deployment/radixjobcomponent.go (hunk 1 above)
Comment (satr) [reply pkg/apis/deployment/radixjobcomponent.go:112]: Fixed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**